src/doc/markdown/page111_developing_with_react_context.md

```diff
@@ -0,0 +1,162 @@
+* Eventually we will migrate `src/web/shared/Utilities.tsx` functions into this directory
+
+# Context Structure
+### `GlobalContext`
+* Contains general data pertaining to the view state of the interface
+* *Examples:*
+    * `selectedPodElement`
+    * `showHubDetails`
+    * `hubAccordionStates`
+#### `PodContext`
+* Encapsulates `HubContext` and `BotContext`
+* Provides a location to store pod-wide data
+##### `HubContext`
+* Stores the `PortalHubStatus`
+##### `BotContext`
+* Coming soon
+
+## General Context Strcuture
+1. Imports
+2. Interfaces
+3. Constant Variables and Enums
+4. Context Instantiation
+5. Reducer Function
+6. Reducer Helper Functions
+7. ContextProvider Component
+
+
+#### 1. Imports
+#### 2. Interfaces
+##### Required Interfaces
+* ContextType
+    * Identifies the properties and types that will be stored in the context
+    * Think of this as the state interface
+```
+// Example
+interface GlobalContextType {
+    clientID: string,
+    controllingClientID: string
+    selectedPodElement: SelectedPodElement
+    ...
+}
+```
+* Action
+    * Lays out the properties and types of the action object passed to the reducer function. `type` is the only required property.
+    * The optional properties are used to pass data to specific reducer helper functions
+```
+// Example
+interface Action {
+    type: string,
+    clientID?: string,
+    hubAccordionName?: string
+}
+```
+* ContextProviderProps
+    * Satisfies TypeScript's requirements for accepting a React Component as a prop
+```
+// Example
+interface GlobalContextProviderProps {
+    children: ReactNode
+}
+```
+
+#### 3. Constant Variables and Enums
+#### 4. Context Instantiation
+The default values of the context are set to `null` because they are set in the context provider function. This convention is discussed in the React documentation: (https://react.dev/reference/react/createContext).
+
+We use two instances of createContext. The first is responsible for holding state and the second is responisble for triggering state changes. This follows the reccommended structure in the React documentation: (https://react.dev/learn/scaling-up-with-reducer-and-context).
+```
+export const GlobalContext = createContext(null)
+export const GlobalDispatchContext = createContext(null)
+```
+
+#### 5. Reducer Function
+A reducer function is called by a dispatch function that comes from "using" the dispatch context. The only required property of the `action` object is `type` which is a string describing the type of action to dispatch. In some cases, you will want to pass data to the reducer function to set state to your desired value. If that is the case, modify the `Action` interface, so you can pass that data in with the `action` object.   
+```
+function globalReducer(state: ContextType, action: Action) {
+    let mutableState = {...state}
+    switch (action.type) {
+        case 'SAVED_CLIENT_ID':
+            return saveClientID(mutableState, action.clientID)
```

Comment:
I can add some more documentation around this. Essentially if you need a piece of data to update state the way you want, you would append an optional property to the action interface and save the data to that property. Only the case (in the switch statement) for the action.type you call will be looking for that property which is why I did not use a question mark because that case is expecting that data.

Reply:
This is an interesting point.  We'll probably have to document which fields are required for each action type.  Preferably, IMO, in a way where it'll show up by hovering in an IDE like VSCode.